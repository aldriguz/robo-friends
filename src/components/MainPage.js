import React, { Component } from 'react';
import CardList from '../components/CardList';
import SearchBox from '../components/SearchBox';
import Scroll from '../components/Scroll';
import ErrorBoundry from '../components/ErrorBoundry';
import Header from '../components/Header';

import './MainPage.css';

class MainPage extends Component {
    componentDidMount() {        
        this.props.onRequestRobots();
    }

    filterRobots() {
        return this.props.robots.filter(robot =>{ 
            return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase());
        });
    }

    render(){
        const {onSearchChange, isPending, error} = this.props;
        
        if(!isPending && error !== '')
            return <h1>An error has ocurred</h1>

        return isPending ?
            <h1>Loading</h1> : 
           (               
                <div className='tc'>
                    <Header />
                    <SearchBox searchChange={onSearchChange}/>
                    <Scroll>
                        <ErrorBoundry>
                            <CardList robots={this.filterRobots()} />
                        </ErrorBoundry>
                    </Scroll>
                </div>
            );
    }
}

export default MainPage;
