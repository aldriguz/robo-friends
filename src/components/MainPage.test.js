import { shallow } from 'enzyme';
import React from 'react';
import MainPage from './MainPage';

let wrapper;

beforeEach(() => {

    const mockStore = {
        onRequestRobots: jest.fn(),
        searchField: '',
        robots: [],
        isPending: false
    };
    
    wrapper = shallow(<MainPage { ...mockStore}/>);
});

it('renders MainPage without crash', () => {
    expect(wrapper).toMatchSnapshot();
})

it('filter robots correctly', () => {
    expect(wrapper.instance().filterRobots()).toEqual([]);
})

it('filter robots correctly with param ', () => {
    const mockStore = {
        onRequestRobots: jest.fn(),
        searchField: 'john',
        robots: [{
            id: 3,
            name: 'John',
            email: 'john@email.com'
        }],
        isPending: false
    };

    wrapper = shallow(<MainPage { ...mockStore}/>);

    expect(wrapper.instance().filterRobots()).toEqual([{
        id: 3,
        name: 'John',
        email: 'john@email.com'
    }]);
})

it('filter robots correctly with param ', () => {
    const mockStore = {
        onRequestRobots: jest.fn(),
        searchField: 'john',
        robots: [],
        isPending: true
    };

    wrapper = shallow(<MainPage { ...mockStore}/>);
    expect(wrapper.contains(<h1>Loading</h1>)).toEqual(true);
})