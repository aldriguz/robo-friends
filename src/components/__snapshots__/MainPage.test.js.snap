// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MainPage without crash 1`] = `
<h1>
  An error has ocurred
</h1>
`;
